src/main/java/frc/robot/commands/AlignToPose.java

```diff
@@ -0,0 +1,133 @@
+import frc.robot.Constants;
+import frc.robot.subsystems.Subsystems;
+import frc.robot.subsystems.Swerve;
+
+/**
+ * A {@link Command} that autonomous drives and aligns the robot to the specified branch of the
+ * nearest reef side.
+ */
+@RobotPreferencesLayout(groupName = "AlignToPose", row = 0, column = 4, width = 2, height = 3)
+public class AlignToPose extends Command {
+  private static final DataLog LOG = DataLogManager.getLog();
+  private static final double MAX_TRANSLATIONAL_POWER = 0.30;
+  private static final double MAX_ROTATIONAL_POWER = 0.5;
+
+  @RobotPreferencesValue public static DoubleValue Px = new DoubleValue("AlignToReef", "X KP", 1);
+  @RobotPreferencesValue public static DoubleValue Py = new DoubleValue("AlignToReef", "Y KP", 1);
+
+  @RobotPreferencesValue
+  public static DoubleValue Pr = new DoubleValue("AlignToReef", "R KP", 0.02);
+
+  protected final Swerve drivetrain;
+
+  protected Pose2d targetPose;
+
+  private final PIDController xController = new PIDController(Px.getValue(), 0, 0);
+  private final PIDController yController = new PIDController(Py.getValue(), 0, 0);
+  private final PIDController rController = new PIDController(Pr.getValue(), 0, 0);
+
+  private final DoubleLogEntry xErrorLog = new DoubleLogEntry(LOG, "Vision/Reef X Error");
+  private final DoubleLogEntry yErrorLog = new DoubleLogEntry(LOG, "Vision/Reef Y Error");
+  private final DoubleLogEntry rErrorLog = new DoubleLogEntry(LOG, "Vision/Reef Yaw Error");
+
+  private final DoubleLogEntry xTargetLog = new DoubleLogEntry(LOG, "Vision/Reef Target X");
+  private final DoubleLogEntry yTargetLog = new DoubleLogEntry(LOG, "Vision/Reef Target Y");
+  private final DoubleLogEntry rTargetLog = new DoubleLogEntry(LOG, "Vision/Reef Target Yaw");
+
+  private double xTarget;
+  private double yTarget;
+  private double rTarget; // in degrees
+
+  /** Creates a new {@link AlignToReef} command. */
+  public AlignToPose(Subsystems subsystems) {
+    this.drivetrain = subsystems.drivetrain;
+
+    // Declare subsystem dependencies.
+    addRequirements(drivetrain);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    System.out.println("TARGET pose: " + targetPose);
+
+    xTarget = targetPose.getX();
+    yTarget = targetPose.getY();
+    rTarget = targetPose.getRotation().getDegrees();
+
+    xTargetLog.append(xTarget);
+    yTargetLog.append(yTarget);
+    rTargetLog.append(rTarget);
+
+    xController.setSetpoint(xTarget);
+    yController.setSetpoint(yTarget);
+    rController.setSetpoint(rTarget);
+
+    xController.setPID(Px.getValue(), 0, 0);
+    yController.setPID(Py.getValue(), 0, 0);
+    rController.setPID(Pr.getValue(), 0, 0);
+
+    xController.setTolerance(Constants.VisionConstants.REEF_ALIGNMENT_TOLERANCE_XY);
+    yController.setTolerance(Constants.VisionConstants.REEF_ALIGNMENT_TOLERANCE_XY);
+    rController.setTolerance(Constants.VisionConstants.REEF_ALIGNMENT_TOLERANCE_R);
```

Comment:
Use "POSE" instead of "REEF"

```suggestion
    xController.setTolerance(Constants.VisionConstants.POSE_ALIGNMENT_TOLERANCE_XY);
    yController.setTolerance(Constants.VisionConstants.POSE_ALIGNMENT_TOLERANCE_XY);
    rController.setTolerance(Constants.VisionConstants.POSE_ALIGNMENT_TOLERANCE_R);
```